admin/class-yoast-network-settings-api.php

```diff
@@ -0,0 +1,195 @@
+			$value = wp_unslash( $value );
+
+			WPSEO_Options::update_site_option( $option_name, $value );
+		}
+
+		$settings_errors = get_settings_errors();
+		if ( empty( $settings_errors ) ) {
+			add_settings_error( $option_group, 'settings_updated', __( 'Settings Updated.', 'wordpress-seo' ), 'updated' );
+		}
+
+		// Use a regular transient here, since it is automatically cleared right after the redirect.
+		// A network transient would be cleaner, but would require a lot of copied code from core for
+		// just a minor adjustment when displaying settings errors.
+		set_transient( 'settings_errors', get_settings_errors(), 30 );
+
+		$sendback = add_query_arg( 'settings-updated', 'true', wp_get_referer() );
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Sanitization callback to use instead of the actual callback to the `register_setting()` method.
+	 *
+	 * This callback will ensure validation only happens in the network admin, to prevent conflicts.
+	 *
+	 * @param string $value  The sanitized option value.
+	 * @param string $option The option name.
+	 *
+	 * @return string The filtered sanitized option value.
+	 */
+	public function filter_sanitize_option( $value, $option ) {
+
+		if ( empty( $this->registered_settings[ $option ] ) ) {
+			return $value;
+		}
+
+		return call_user_func( $this->registered_settings[ $option ]['santize_callback'], $value );
+	}
+
+	/**
+	 * Filters the default value for a network option.
+	 *
+	 * This function is added as a filter to `default_site_option_{$option}` for network options that
+	 * are registered with a default.
+	 *
+	 * @param mixed  $default Existing default value to return.
+	 * @param string $option  The option name.
+	 *
+	 * @return mixed The filtered default value.
+	 */
+	public function filter_default_option( $default, $option ) {
+
+		// If a default value was manually passed to the function, allow it to override.
+		if ( $default !== false ) {
+			return $default;
+		}
+
+		if ( empty( $this->registered_settings[ $option ] ) ) {
+			return $default;
+		}
+
+		return $this->registered_settings[ $option ]['default'];
+	}
+
+	/**
+	 * Hooks in the necessary actions and filters.
+	 */
+	public function register_hooks() {
+
+		if ( $this->hooks_registered ) {
+			return;
+		}
+
+		$this->hooks_registered = true;
+
+		add_action( 'admin_action_' . self::UPDATE_ACTION, array( $this, 'handle_update_options_request' ) );
+	}
+
+	/**
+	 * Gets the singleton instance of this class.
+	 *
+	 * @return Yoast_Network_Settings_API The singleton instance.
+	 */
+	public static function get() {
+
+		if ( self::$instance === null ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Checks the requirements for this class to be used.
+	 *
+	 * If these requirements are not met, this class should not be initialized.
+	 *
+	 * @return bool True if requirements are met, false otherwise.
+	 */
+	public static function check_requirements() {
```

Comment:
This function name seems misleading to me. I'd consider renaming it to something such as `meets_requirements`, which makes the usage of the it in if-statements, much more readable.